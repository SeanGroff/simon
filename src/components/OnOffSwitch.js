import React from 'react'
import styled from 'styled-components'

import { useGame } from '../hooks/useGame'

const StyledSwitch = styled.div`
  width: 40px;
  height: 20px;
  background-color: #222222;

  :hover {
    cursor: pointer;
  }
`

const Toggler = styled.div`
  background-color: #fc0102;
  width: 16px;
  height: 16px;
  border-radius: 4px;
  border: 2px solid #222222;
  position: relative;
  left: ${({ power }) => (power ? '20px' : '0')};
  transition: left 0.1s ease-out;
`

export default function OnOffSwitch() {
  const [state, dispatch, actions] = useGame()
  const handleClick = () => {
    dispatch({
      type: actions.TOGGLE_GAME_POWER,
      power: !state.power,
    })
  }
  return (
    <StyledSwitch onClick={handleClick}>
      <Toggler power={state.power} />
    </StyledSwitch>
  )
}
