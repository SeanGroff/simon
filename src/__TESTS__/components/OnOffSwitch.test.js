import React from 'react';
import { shallow } from 'enzyme';
import OnOffSwitch from '../../components/OnOffSwitch';

it('renders a toggle switch correctly', () => {
  const component = shallow(<OnOffSwitch />);
  expect(component).toMatchSnapshot();
});

// it('should contain <Toggler />', () => {
//   const component = shallow(<OnOffSwitch />);
//   const toggler = component.find('Toggler');
//   expect(toggler.length).toBe(1);
// });
