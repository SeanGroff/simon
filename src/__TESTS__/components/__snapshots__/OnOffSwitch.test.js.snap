// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a toggle switch correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <styled.div>
    <styled.div
        off={true}
        onClick={false}
    />
</styled.div>,
  "nodes": Array [
    <styled.div>
      <styled.div
            off={true}
            onClick={false}
      />
</styled.div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <OnOffSwitch
        off={true}
/>,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "off": true,
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <styled.div>
          <styled.div
                    off={true}
                    onClick={false}
          />
</styled.div>,
        "_debugID": 2,
        "_renderedOutput": <styled.div>
          <styled.div
                    off={true}
                    onClick={false}
          />
</styled.div>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <OnOffSwitch
    off={true}
/>,
}
`;
